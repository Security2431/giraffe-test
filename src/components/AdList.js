import React from 'react';
import PropTypes from 'prop-types';
import AdListItem from './AdListItem';
import AdListPagination from './AdListPagination';

const defaults = {
  descrFixedLength: 150,
  pageSize: 5,
};

const theme = {
  Col: {
    flex: '0 0 20%',
    maxWidth: '20%',
  },
};

class AdList extends React.Component {
  static propTypes = {
    activeUser: PropTypes.string.isRequired,
    ads: PropTypes.arrayOf(
      PropTypes.shape({
        _id: PropTypes.string.isRequired,
        author_name: PropTypes.string.isRequired,
        created_at_datetime: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
      }),
    ),
    deleteAd: PropTypes.func.isRequired,
  };

  static defaultProps = {
    ads: null,
  };

  state = {
    currentPage: null,
  };

  currentPage = (page) => {
    this.setState({
      currentPage: page,
    });
  };

  render() {
    const { ads, activeUser, deleteAd } = this.props;
    const { currentPage } = this.state;
    const { descrFixedLength, pageSize } = defaults;

    const startIndex = (currentPage - 1) * pageSize;
    const endIndex = startIndex + pageSize;

    const adList = ads
      .filter((item, index) => index >= startIndex && index < endIndex)
      .map((item) => {
        const { _id: id } = item;

        return (
          <div className="col-6 col-sm-4 col-lg-3 col-xl" style={theme.Col} key={id}>
            <AdListItem
              activeUser={activeUser}
              descrFixedLength={descrFixedLength}
              deleteAd={deleteAd}
              {...item}
            />
          </div>
        );
      });

    return (
      <React.Fragment>
        {adList}
        <div className="col-12 d-flex justify-content-center">
          <AdListPagination
            pageSize={pageSize}
            adsCount={ads.length}
            currentPage={this.currentPage}
          />
        </div>
      </React.Fragment>
    );
  }
}

export default AdList;
