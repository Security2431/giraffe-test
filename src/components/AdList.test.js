/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AdList from './AdList';

jest.mock('./AdListItem', () => () => 'AdListItem');
jest.mock('./AdListPagination', () => () => 'AdListPagination');

const mockProps = {
  deleteAd: jest.fn(),
  activeUser: 'Admin',
  ads: [
    {
      _id: 'cjv7wpcgj00022y5rtui74uls',
      author_name: 'Admin',
      created_at_datetime: '2019-05-03T12:59:23+03:00',
      title: 'Title',
      description: 'Hello, World!',
    },
    {
      _id: 'cjv7wv7cv00032y5r9yyvmj2x',
      author_name: 'Artem',
      created_at_datetime: '2019-05-03T12:59:23+03:00',
      title: 'Title 2',
      description: 'Hello, React!',
    },
  ],
};

describe('<AdList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AdList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AdList {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
