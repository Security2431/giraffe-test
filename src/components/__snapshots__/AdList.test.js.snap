// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdList /> renders correctly 1`] = `
<div
  className="col-12 d-flex justify-content-center"
>
  AdListPagination
</div>
`;
